gapic/src/main/com/google/gapid/image/ArrayImage.java

```diff
@@ -302,47 +293,38 @@ protected void convert2D(byte[] src, byte[] dst, byte[] alpha, int stride) {
     }
 
     @Override
-    protected PixelValue getPixel(int x, int y, byte[] data) {
+    protected PixelValue getPixel(int x, int y) {
       int i = 4 * (y * width + x);
       return new Pixel(buffer.get(i + 0), buffer.get(i + 1), buffer.get(i + 2), buffer.get(i + 3));
     }
 
     @Override
-    public Set<Channel> getChannels() {
-      return Sets.immutableEnumSet(Channel.Red, Channel.Green, Channel.Blue, Channel.Alpha);
+    public Set<Stream.Channel> getChannels() {
+      return Images.RGB_CHANNELS;
     }
 
     @Override
-    public DoubleBuffer getChannel(Channel channel) {
-      int count = width * height * depth;
-      DoubleBuffer out = DoubleBuffer.allocate(count);
-      int offset = 0;
-      switch (channel) {
-        case Red:
-          offset = 0;
-          break;
-        case Green:
-          offset = 1;
-          break;
-        case Blue:
-          offset = 2;
-          break;
-        case Alpha:
-          offset = 3;
-          break;
-        default:
-          return out;
-      }
-      for (int i = 0; i < count; i++) {
-        out.put(buffer.get(i * 4 + offset));
-      }
-      out.rewind();
-      return out;
+    public boolean isHDR() {
+      return true;
     }
 
     @Override
-    public boolean isHDR() {
-      return true;
+    public void bin(Histogram.Binner binner) {
```

Comment:
I'm not sold on this - replacing a very useful 'give me the value of this pixel in a standardized format' with something so heavily tied to the histogram seems like an overall loss. Given that you're calling back to the binner for each pixel component, I can't imagine it's all that much faster.

Reply:
Fair enough. I can't argue with your profiling results. I guess if we need to sample a pixel, we can always add that back in again.